### PR TITLE
changedURI use defer seq

### PR DESCRIPTION
**Pull request by yingjieMiao**
@leogrim @raymondng

### CODE REVIEW

**Comment by leogrim** [comment]
:+1:

File: kifi-backend/modules/shoebox/test/com/keepit/integrity/OrphanCleanerTest.scala [hunk 1]
```diff
@@ -106,6 +106,8 @@ class OrphanCleanerTest extends Specification with ShoeboxTestInjector {
         db.readWrite { implicit session =>
           uris.drop(1).map { uri => changedURIRepo.save(ChangedURI(oldUriId = uri.id.get, newUriId = uris(0).id.get, state = ChangedURIStates.APPLIED)) }
         }
+        val seqAssigner = inject[ChangedURISeqAssigner]
+        seqAssigner.assignSequenceNumbers()
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/test/com/keepit/integrity/OrphanCleanerTest.scala:110]: Sorry, a bit unclear to me. What does assigning sequence numbers in tests do?

Comment (leogrim) [inline kifi-backend/modules/shoebox/test/com/keepit/integrity/OrphanCleanerTest.scala:110]: Previously, the sequence number would be assigned immediately on `save`. Now if your test does rely on sequence numbers, you want to make sure they are indeed assigned by the assigner (vs hoping for the asynchronous plugin to do it).

Comment (ahsu1230) [inline kifi-backend/modules/shoebox/test/com/keepit/integrity/OrphanCleanerTest.scala:110]: I see thanks!

**Pull request merged**